pkg/cloudprovider/drift.go

```diff
@@ -39,58 +40,61 @@ import (
 )
 
 const (
-	NodeClassDrift  cloudprovider.DriftReason = "NodeClassDrift"
-	K8sVersionDrift cloudprovider.DriftReason = "K8sVersionDrift"
-	ImageDrift      cloudprovider.DriftReason = "ImageDrift"
-	SubnetDrift     cloudprovider.DriftReason = "SubnetDrift"
+	NodeClassDrift       cloudprovider.DriftReason = "NodeClassDrift"
+	K8sVersionDrift      cloudprovider.DriftReason = "K8sVersionDrift"
+	ImageDrift           cloudprovider.DriftReason = "ImageDrift"
+	SubnetDrift          cloudprovider.DriftReason = "SubnetDrift"
+	KubeletIdentityDrift cloudprovider.DriftReason = "KubeletIdentityDrift"
 
 	// TODO (charliedmcb): Use this const across code and test locations which are signaling/checking for "no drift"
 	NoDrift cloudprovider.DriftReason = ""
 )
 
 func (c *CloudProvider) isNodeClassDrifted(ctx context.Context, nodeClaim *karpv1.NodeClaim, nodeClass *v1beta1.AKSNodeClass) (cloudprovider.DriftReason, error) {
-	// First check if the node class is statically staticFieldsDrifted to save on API calls.
-	if staticFieldsDrifted := c.areStaticFieldsDrifted(nodeClaim, nodeClass); staticFieldsDrifted != "" {
-		return staticFieldsDrifted, nil
+	// TODO: if we find more expensive checks, such as reading VMs or NICs from Azure, are being duplicated between checks, we should
+	// TODO: produce a lazy at-most-once that allows a check to cache a value for later checks to read.
```

Comment:
Nit: I like leaving it just so one of the lines has a TODO in it, with indent, so searching only leaves with one occurrence showing up, but just my feeling of it.
```suggestion
	//     produce a lazy at-most-once that allows a check to cache a value for later checks to read.
```

Reply:
Fixed